apis/workloads/v1alpha1/replicatedstatemachine_webhook.go

```diff
@@ -80,24 +80,28 @@ func (r *ReplicatedStateMachine) ValidateDelete() error {
 func (r *ReplicatedStateMachine) validate() error {
 	var allErrs field.ErrorList
 
-	// Leader is required
-	hasHeader := false
-	for _, role := range r.Spec.Roles {
-		if role.IsLeader && len(role.Name) > 0 {
-			hasHeader = true
+	if len(r.Spec.Roles) > 0 {
+		// Leader is required
+		hasHeader := false
+		for _, role := range r.Spec.Roles {
+			if role.IsLeader && len(role.Name) > 0 {
+				hasHeader = true
```

Comment:
It's better to add a `break`

Reply:
done: [09375c2](https://github.com/apecloud/kubeblocks/pull/4086/commits/09375c28277be20a9780721a4de775ed38a96f96)